check.sh

```diff
@@ -0,0 +1,76 @@
+#!/bin/sh
+
+try_installing() {
+
+    if [ ! $NO_XBPS ]; then
+        echo "\nTrying to install '$PKG' ... "
+        if [ ! -z $SU_CMD ]; then
+            $SU_CMD xbps-install $PKG
+
+            # 'xbps-install' exits with 0 when the user aborts, so we can't
+            # check return status; use 'which' to check status instead.
+
+            if ( ! which mdbook 2>&- 1>&- ); then
+                echo "Can't proceed without '$PKG', aborting."
+                exit 1
+            fi
+        else
+            "Can't find 'sudo' or 'doas'; please install '$PKG' manually."
+            exit 1
+        fi
+    else
+        echo "Can't find 'xbps-install'; please install '$PKG' manually."
+        exit 1
+    fi
+
+}
+
+echo -n "Checking for mdbook ... "
+if ( ! which mdbook 2>/dev/null ); then
```

Comment:
I prefer to use `command -v` instead of `which`, since it's a shell built-in.

Reply:
i was having difficulties controlling the output of `command -v` as i wanted, particularly when checking for `xbps-install`, for which i didn't see the need to have a "Checking ..." line. But i suppose having the latter would be a good idea, so i'll change all uses of `which` to `command -v`.